src/core/mac/mac.cpp

```diff
@@ -2061,5 +2061,69 @@ uint8_t Mac::GetTimeIeOffset(const Frame &aFrame)
 }
 #endif
 
+#if OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
+otError Mac::AppendHeaderIe(Message *aMessage, TxFrame &aFrame) const
```

Comment:
I would suggest not using `Message *aMessage` as input for these new methods
- `Mac` generally deals with `Frame` and is not aware of `Message` (which are handled by next layer `MeshForwarder`).
- Seems like `aMessage` is only used to detect if it is time-sync or not `aMessage->IsTimeSync()`.

How about adding a `bool aIsTimeSync` as input parameter to this method instead?